### ISSUE TITLE
the rusult"output" is error.

### ISSUE THREAD

**Issue by fenling** [agethen/ConvLSTM-for-Caffe#5]
hi agethen:
     i use your "encode-decode.prototxt", max_iter: 20000. the train image size is [20,16,16,16,16],the num    is   625. the 20000' cross_entropy_loss is 9806.
     when i check the output ,i find the output image is error.
     the input data range is [0,1] ，but the output data range is [-35.96~1.91].
     the output range should also be[0,1]?

**Comment by agethen** [comment]
Dear fenling,

the blob "output" may indeed be in a different range than [0,1]. To understand that, you need to understand the "SigmoidCrossEntropy" of Caffe. As the name says, it computes a Sigmoid (only on the first input, NOT on the second input).

Now consider that "output" were already in [0,1]. If the CrossEntropy reapplies the sigmoid, that 'squeezes' the values into something like 0.5 and 0.73. Therefore, we cannot do that.

If you want to plot the output, you simply need to pass it through a "Sigmoid" layer. Check the example near the end on how to do that: The blob "out_sigm" is exactly that (however, it is flattened, and I think you want the 2D image).

**Comment by EllieLily** [comment]
Hi  fenling:
      did you get the final output images?
      I tried to train this model,but just got some blurred pictures. did you get images like this too?
     
![32139942-14862faa-bc8b-11e7-9d7a-fef0bb7fb093](https://user-images.githubusercontent.com/6090769/34604722-d6d27648-f243-11e7-8404-99355371a097.png)
![32139936-f334a048-bc8a-11e7-8254-5983edf5b4b8](https://user-images.githubusercontent.com/6090769/34604816-256edb3e-f244-11e7-8b96-9e16d38b472f.png)
